### ISSUE TITLE
Not an issue: post login hook to assign users to groups

### ISSUE THREAD

**Issue by dixnor** [finalist/liferay-oidc-plugin#3]
This is not an issue but more information request on which class the plugin provides to be able to achieve what we are trying to do.
We are using this very usefull plugin to register and login user coming from an OpenID authentication provider. 
When requesting the user profile with the OpenID token, some claims are also provided. We would like to be able to assign users to preexisting groups in Liferay based on these claims.
So basically wanted to know how you would do that post-login logic and on which class we should hook to be able to retrieve these claims.
Asking this question based on marketplace plugin description when you say:

Can be combined with post login hooks to perform additional custom logic (adding users to groups etc)

Thanks for providing some hints.

**Comment by gvanderploeg** [comment]
Hi Dixnor,

Great to hear that the plugin works for your use case.
In one of our projects, we actually use a post login hook to grant usergroup membership based on the OpenID attributes, so that's definitely a working option.

The OpenID Connect filter puts the information received from the OpenID flow into a session object, and this info can be retrieved by a post login action.

You can make use of this yourself with a snippet like this:
`Map<String, String> userInfo = (Map<String, String>) session.getAttribute(
                LibFilter.OPENID_CONNECT_SESSION_ATTR);`

The actual content of the resulting map depends on the Authorization server you use, and the scopes that you have defined/granted, so you could start with enabling logging on debug level for the OpenID Connect filter (category `nl.finalist.liferay.oidc`) that will output the contents of the map as it is put into the session.

This is not a complete code sample of course, but hopefully this helps you create your own post login hook now.

**Comment by dixnor** [comment]
Hi Dvanderploeg,
Thanks for your fast answer. The plugin works very well indeed in our case.
We will give a try to the approach you pointed us and hoppefully will be able to achieve what you did.

**Comment by dixnor** [comment]
Finally we have been able to figure out, using your snippet,  how to retrieve correctly scopes from the token and use them to assign groups membership. Thanks for you help.

This issue can been closed now.
